.github/workflows/macos-cmake.yml

```diff
@@ -37,12 +46,68 @@ jobs:
         create_credentials_file: true
         credentials_json: ${{ secrets.BUILD_CACHE_KEY }}
     - uses: google-github-actions/setup-gcloud@v1
-    - name: Get External Configuration
-      id: info
+    - name: Dynamic Configuration
+      id: dynamic
       shell: bash
       run: |
         echo "vcpkg-version=$(cat ci/etc/vcpkg-version.txt)" >> "${GITHUB_OUTPUT}"
-        echo "features=$(jq -c .${{ matrix.shard }} < .github/workflows/macos-cmake-features.json)" >> "${GITHUB_OUTPUT}"
+        core_features=(
+          bigtable
+          pubsub
+          pubsublite
+          spanner
+          storage
+        )
+        # These are the libraries with the most "clients". To build the list
+        # run something like this and find the midpoint:
+        #
+        # git grep -l 'class.*Client' 'google/cloud/**_client.h' |
+        #    egrep -v "(bigtable/|internal/|pubsub/|spanner/|storage/)" |
+        #    cut -f -3 -d/| sort | uniq -c | sort -n |
+        #    awk '{ s += $1; print s, $0}'
+        #
+        shard1_features=(
+          servicecontrol
+          support
+          video
+          datacatalog
+          iam
+          kms
+          beyondcorp
+          contentwarehouse
+          dataproc
+          talent
+          retail
+          bigquery
+          appengine
+          monitoring
+          sql
+          dialogflow_cx
+          aiplatform
+          dialogflow_es
+        )
+        if [[ "${{ matrix.shard }}" == "Core" ]]; then
+          features="$(printf ",%s" "${core_features[@]}")"
+          echo "features=${features:1}" >> "${GITHUB_OUTPUT}"
+        elif [[ "${{matrix.shard}}" == "Compute" ]]; then
+          echo "features=compute" >> "${GITHUB_OUTPUT}"
+        elif [[ "${{ matrix.shard }}" == "Shard1" ]]; then
+          features="$(printf ",%s" "${shard1_features[@]}")"
+          echo "features=${features:1}" >> "${GITHUB_OUTPUT}"
+        else
+          skipped_features=("${core_features[@]}")
+          skipped_features+=(compute)
+          skipped_features+=("${shard1_features[@]}")
+          # Disable `storagetranster` on macOS+CMake because:
+          # - we use vcpkg in this build
+          # - vcpkg ships with Protobuf v21
+          # - GID_MAX is generated in the *.pb.h file
+          # - GID_MAX is a system macro on macOS
+          # - Protobuf v21 does not protect against this macro
```

Comment:
These issues and protobuf v21 are things of the past, yes?

Reply:
The build lives in the past.... it uses vcpkg, and vcpkg has not upgraded to Protobuf v23.  They are working on it:

https://github.com/microsoft/vcpkg/pull/32602